### PR TITLE
Improve expected test errors for some old cases

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/libsolidity/SolidityNameAndTypeResolution.cpp [hunk 1]
```diff
@@ -4033,7 +4033,7 @@ BOOST_AUTO_TEST_CASE(varM_disqualified_as_keyword)
 			}
 		}
 	)";
-	BOOST_CHECK(!success(text));
+	CHECK_ERROR(text, DeclarationError, "Identifier not found or not unique.");
```
Comment (axic) [inline test/libsolidity/SolidityNameAndTypeResolution.cpp:4036]: This is what it currently outputs, but shouldn't it be a parser error?

File: test/libsolidity/SolidityNameAndTypeResolution.cpp [hunk 2]
```diff
@@ -4074,7 +4074,7 @@ BOOST_AUTO_TEST_CASE(long_uint_variable_fails)
...
```
Comment (axic) [inline test/libsolidity/SolidityNameAndTypeResolution.cpp:4077]: Same here.

File: test/libsolidity/SolidityNameAndTypeResolution.cpp (hunk 1 above)
Comment (chriseth) [reply test/libsolidity/SolidityNameAndTypeResolution.cpp:4036]: `uintM` is not a keyword, so it parses it as the name of a type. The error class is fine I woulds ay.

**Review by chriseth — APPROVED** [review]

File: test/libsolidity/SolidityNameAndTypeResolution.cpp (hunk 1 above)
Comment (axic) [reply test/libsolidity/SolidityNameAndTypeResolution.cpp:4036]: Ah right. I just would rather not have native types in the parser to avoid confusion.

**Comment by axic** [comment]
Has the usual travis failure for http.

**Pull request merged**